pkg/services/ngalert/api/api_ruler.go

```diff
@@ -259,6 +260,99 @@ func (srv RulerSrv) RouteGetRulesGroupConfig(c *contextmodel.ReqContext, namespa
 	return response.JSON(http.StatusAccepted, result)
 }
 
+// return list of rule groups paginated
+func (srv RulerSrv) RouteGetAlertRuleGroups(c *contextmodel.ReqContext) response.Response {
+	namespaceMap, err := srv.store.GetUserVisibleNamespaces(c.Req.Context(), c.SignedInUser.GetOrgID(), c.SignedInUser)
+	if err != nil {
+		return ErrResp(http.StatusInternalServerError, err, "failed to get namespaces visible to the user")
+	}
+
+	result := apimodels.NamespaceListResponse{
+		RuleGroups: make([]apimodels.RuleGroupSummary, 0),
+	}
+
+	if len(namespaceMap) == 0 {
+		srv.log.Debug("User has no access to any namespaces")
+		return response.JSON(http.StatusOK, result)
+	}
+
+	namespaceUIDs := make([]string, len(namespaceMap))
+	for k := range namespaceMap {
+		namespaceUIDs = append(namespaceUIDs, k)
+	}
+
+	const defaultPageSize = 100
+	var namespaceListParams apimodels.NamespaceListParams
+	namespaceListParams.NextToken = c.Query("group_next_token")
+	if namespaceListParams.NextToken != "" {
+		// make sure it base64 decodes
+		if _, err := base64.URLEncoding.DecodeString(namespaceListParams.NextToken); err != nil {
+			return ErrResp(http.StatusBadRequest, err, "invalid next_token")
+		}
+	}
+	namespaceListParams.PageSize = c.QueryIntWithDefault("group_limit", defaultPageSize)
+
+	// NOTE: can probably use HasAccessInFolder in the inner loop to avoid fetching all rules
+	// TODO: clean this to ignore the rule bits and make sure it's returned sorted
+	configs, _, err := srv.searchAuthorizedAlertRules(c.Req.Context(), authorizedRuleGroupQuery{
+		User:          c.SignedInUser,
+		NamespaceUIDs: namespaceUIDs,
+	})
+	keys := make([]ngmodels.AlertRuleGroupKey, 0, len(configs))
+
+	// in-place sort by namespaceUID and ruleGroup
```

Comment:
We want to sort by the Namespace itself rather than its UID

Reply:
We don't have a namespace at this point per say, but I could try to use the full folder path. Does that work?

Reply:
Yeah, in this context `namespace` is the full folder path 😄